src/Transform/ONNX/ONNXDimAnalysis.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------- ONNXDimAnalysis.hpp - ONNX Dimension Analysis ---------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file implements an analysis on unknown dimensions in ONNX ops.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace onnx_mlir {
+
+class DimAnalysis {
+public:
+  using DimT = std::pair<mlir::Value, uint64_t>;
```

Comment:
Can you indicate for each type what they are used for. That would help the comprehension.

Reply:
tx